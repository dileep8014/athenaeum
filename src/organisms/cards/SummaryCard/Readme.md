### Split Card Example:

```jsx
    <SummaryCard
      text={[
        {
          header: 'header',
          subheader: 'subheader'
        },
        {
          header: 'secondaryHeader',
          subheader: 'another subhead'
        }
      ]}
      icon='xIcon'
      onIconClick={() => alert('you clicked real good')}
      footerLinks={[
        {
          text: 'A link',
          onClick: () => alert('footer')
        },
        {
          text: 'second link',
          onClick: () => alert('second')
        }
      ]}
    />
```

```jsx
    <SummaryCard
      text={[
        {
          header: 'header',
          subheader: 'subheader'
        },
      ]}
      icon='xIcon'
      onIconClick={() => alert('you clicked real good')}
      footerLinks={[
        {
          text: 'A link',
          onClick: () => alert('footer')
        },
        {
          text: 'second link',
          onClick: () => alert('second')
        }
      ]}
    >
      <Text>
        Hey there how is this going pretty great
      </Text>
    </SummaryCard>
```

```jsx
    <SummaryCard
      text={[
        {
          header: 'header',
          subheader: 'subheader'
        },
        {
          header: 'secondaryHeader',
          subheader: 'another subhead'
        }
      ]}
      icon='greenCheckWithCircle'
      footerLinks={[
        {
          text: 'A link',
          onClick: () => alert('footer')
        },
      ]}
    />
```
