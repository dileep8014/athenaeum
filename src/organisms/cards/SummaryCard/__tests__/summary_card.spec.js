import React from 'react';
import renderer from 'react-test-renderer';

import SummaryCard from 'organisms/cards/SummaryCard';

describe('<SummaryCard />', () => {
  let props;

  beforeEach(() => {
    props = {
      text: [
        { header: 'header', subheader: 'subheader' }, { header: 'secondaryHeader', subheader: 'another subhead' }
      ],
      icon: 'xIcon',
      onIconClick: jest.fn(),
      footerLinks: [
        { text: 'A link', onClick: jest.fn() }, { text: 'second link', onClick: jest.fn() }
      ]
    };
  });

  it('renders correctly', () => {
    const actual = renderer.create(<SummaryCard {...props} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
