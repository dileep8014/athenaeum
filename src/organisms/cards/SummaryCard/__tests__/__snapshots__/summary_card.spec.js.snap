// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryCard /> renders correctly 1`] = `
<div
  className="summary-card"
>
  <div
    className="content"
  >
    <div
      className="layout"
      id={undefined}
      style={undefined}
    >
      <div
        className="col col-sm-8 border-neutral-6"
        onClick={undefined}
        style={undefined}
      >
        <div>
          <p
            className="primary-3 type-a-5-bold inherit"
          >
            header
          </p>
          <div
            className="size-6"
          />
          <p
            className="neutral-3 type-b-8-medium inherit"
          >
            subheader
          </p>
        </div>
      </div>
      <div
        className="col col-sm-4 border-neutral-6"
        onClick={undefined}
        style={undefined}
      >
        <div>
          <p
            className="primary-3 type-a-5-bold inherit"
          >
            secondaryHeader
          </p>
          <div
            className="size-6"
          />
          <p
            className="neutral-3 type-b-8-medium inherit"
          >
            another subhead
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="layout footer"
    id={undefined}
    style={undefined}
  >
    <div
      className="col col-sm-8 border-neutral-6"
      onClick={undefined}
      style={undefined}
    >
      <a
        className="link neutral-1 secondary"
        onClick={[Function]}
      >
        <p
          className="primary-3 type-b-10-medium inherit"
        >
          A link
        </p>
      </a>
    </div>
    <div
      className="col col-sm-4 border-neutral-6"
      onClick={undefined}
      style={undefined}
    >
      <a
        className="link neutral-1 secondary"
        onClick={[Function]}
      >
        <p
          className="primary-3 type-b-10-medium inherit"
        >
          second link
        </p>
      </a>
    </div>
  </div>
</div>
`;
